### PR TITLE
added Getting ROCm Source Files

### PR DESCRIPTION
**Pull request by randyh62**
per SWDEV-449418

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,6 +19,49 @@ ROCm supports programming models, such as OpenMP and OpenCL, and includes all ne
 source software compilers, debuggers, and libraries. ROCm is fully integrated into machine learning
 (ML) frameworks, such as PyTorch and TensorFlow.
 
+## Getting the ROCm Source Code
+
+AMD ROCm is built from open source software. It is, therefore, possible to modify the various components of ROCm by downloading the source code and rebuilding the components. The source code for ROCm components can be cloned from each of the GitHub repositories using git.  For easy access to download the correct versions of each of these tools, the ROCm repository contains a repo manifest file called default.xml. You can use this manifest file to download the source code for ROCm software.
```
Comment (Rmalavally) [inline README.md:24]: Nicely written, Randy. Do we want to add a link to the repo with default.xml?

Comment (randyh62) [reply README.md:24]: Thanks Roopa. I don't think a link is necessary as the default.xml is in the same level of the ROCm repo as the README file.

Comment (Rmalavally) [reply README.md:24]: Thanks for confirming, Randy. Sounds good.

Comment (randyh62) [reply README.md:24]: Thanks for looking.

File: README.md [hunk 2]
```diff
@@ -19,6 +19,49 @@ ROCm supports programming models, such as OpenMP and OpenCL, and includes all ne
...
+
+### Installing the Repo
```
Comment (yhuiYH) [inline README.md:26]:
```suggestion
### Installing the repo tool
```
I think it should be called 'repo tool', even tho the old instructions used the same "Installing the Repo".  But this section is about the repo tool IMO.

Comment (randyh62) [reply README.md]: Updated

File: README.md [hunk 3]
```diff
@@ -19,6 +19,49 @@ ROCm supports programming models, such as OpenMP and OpenCL, and includes all ne
...
+
+Some ROCm projects use the Git Large File Storage (LFS) format that may require you to install git-lfs. Refer to [Git Large File Storage](https://github.com/git-lfs/git-lfs/blob/main/INSTALLING.md) for more information. For example, to install git-lfs for Ubuntu, use the following command: 
+
+```bash
+sudo apt-get install git-lfs
+```
+
+### Downloading the ROCm source code
+
+The following example shows how to use the repo binary to download the ROCm source code. If you choose a directory other than ~/bin/ to install the repo, you must use that chosen directory in the code as shown below:
```
Comment (dgaliffiAMD) [inline README.md:48]:
```suggestion
The following example shows how to use the repo tool to download the ROCm source code. If you choose a directory other than ~/bin/ to install the repo, you must use that chosen directory in the code as shown below:
```

File: README.md [hunk 4]
```diff
@@ -19,6 +19,49 @@ ROCm supports programming models, such as OpenMP and OpenCL, and includes all ne
...
+~/bin/repo sync
+```
+
+**Note:** Using this sample code will cause the repo to download the open source code associated with the specified ROCm release. Ensure that you have ssh-keys configured on your machine for your GitHub ID prior to the download as explained at [Connecting to GitHub with SSH](https://docs.github.com/en/authentication/connecting-to-github-with-ssh).
+
+### Building the ROCm source code
+
+Each ROCm component repository contains directions for building that component, such as the rocSPARSE documentation [Installation and Building for Linux](https://rocm.docs.amd.com/projects/rocSPARSE/en/latest/Linux_Install_Guide.html). Refer to the specific component documentation for instructions on building the repository.
+
+Each release of the ROCm software supports specific hardware and software configurations. Refer to [System requirements (Linux)](https://rocm.docs.amd.com/projects/install-on-linux/en/latest/reference/system-requirements.html) for the current supported hardware and OS.
```
Comment (dgaliffiAMD) [inline README.md:63]:
```suggestion
Each release of the ROCm software supports specific hardware and software configurations. Refer to [System requirements (Linux)](https://rocm.docs.amd.com/projects/install-on-linux/en/latest/reference/system-requirements.html) for the current supported hardware and OS.
```

**Review by dgaliffiAMD — DISMISSED** [review]
Suggested some fixes to linting issues and changed references to "the repo" to "the repo tool". Simply calling it "repo" could work too. 
I'm not sure of all of your formatting conventions, would `repo` be preferred.

File: README.md (hunk 1 above)
Comment (randyh62) [reply README.md]: Hi Roopa, I went ahead and added the link while I was editing other things.

Comment (Rmalavally) [reply README.md]: Excellent! Thanks, Randy.

**Review by yhuiYH — APPROVED** [review]
Now that all linting fixes are resolved, looks good to me.

**Pull request merged**